azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_stages_iothub_mqtt.py

```diff
@@ -137,7 +154,20 @@ def _handle_pipeline_event(self, event):
                 method_received = MethodRequest(
                     request_id=rid, name=method_name, payload=json.loads(event.payload)
                 )
-                self._handle_pipeline_event(pipeline_events_iothub.MethodRequest(method_received))
+                self.handle_pipeline_event(pipeline_events_iothub.MethodRequest(method_received))
+
+            elif mqtt_topic_iothub.is_twin_response_topic(topic):
```

Comment:
It is kinda weird that this has to go from generic (some event) to specific (this is a twin response) to generic (make it a generic IoTResponseEvent) to specific again (resolve the GetTwin operation).

I don't necessarily think this is wrong, nor do I see a better way to do it, but it is weird IMO

Reply:
yes, very weird.

---
In reply to: [288741545](https://github.com/Azure/azure-iot-sdk-python-preview/pull/96#discussion_r288741545) [](ancestors = 288741545)